RIOT-OS/RIOT
# cpu/msp430_common: use malloc_thread_safe

### Contribution description

This should fix concurrent dynamic memory allocation.

### Testing procedure

1. Run `make BOARD=z1 -C tests/malloc_thread_safety flash test` on `master`
2. Same on this PR

It should fail on `master` but get fixed by this PR.

### Issues/PRs references

None
